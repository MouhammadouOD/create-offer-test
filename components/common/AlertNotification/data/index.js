export default {
  message: 'Message',
  type: 'error'
}
