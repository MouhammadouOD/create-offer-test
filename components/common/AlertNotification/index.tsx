import React, { ReactElement, useEffect, useState } from 'react'
import { BiCheck, BiX } from 'react-icons/bi'

type Props = {
  message: string | null
  type: TypeNotification
  handleNotification?: (
    message: string | null,
    typeNotif: TypeNotification
  ) => void
}

const AlertNotification = ({
  message,
  type,
  handleNotification
}: Props): ReactElement => {
  const [displayAlert, setDisplayAlert] = useState<'flex' | 'hidden'>('hidden')
  const [notification, setNotification] = useState<string | null>(message)
  const [typeNotification, setTypeNotification] = useState<string | null>(
    type === 'error' ? 'red' : 'green'
  )
  const baseCSS = 'inline-block mb-1 h-6 w-6'

  useEffect(() => {
    setDisplayAlert(message ? 'flex' : 'hidden')
    setNotification(message ? message : null)
    setTypeNotification(type === 'error' ? 'red' : 'green')
    return () => {
      setDisplayAlert('hidden')
      setNotification(null)
    }
  }, [message, type])

  const removeAlert = () => {
    handleNotification && handleNotification('', 'error')
  }

  return (
    <>
      <div
        data-testid='displayAlert'
        className={`fixed bottom-0 right-0 z-50 ${displayAlert} flex-row justify-between items-center cursor-pointer rounded-md px-4 py-3 my-2 md:px-6 md:py-4 md:my-4 mx-auto text-sm md:text-base 2xl:text-lg w-4/5 lg:w-2/5 2xl:w-2/5  bg-${typeNotification}-200 `}
      >
        <div className='w-full flex flex-row items-center'>
          <span className='w-1/12'>
            {typeNotification === 'red' ? (
              <BiX
                data-testid='typeNotification1'
                className={`${baseCSS} text-red-600`}
              />
            ) : (
              <BiCheck className={`${baseCSS} text-green-600`} />
            )}
          </span>
          <span
            data-testid='typeNotification2'
            className={`ml-2 inline-block w-11/12 text-${typeNotification}-800`}
          >
            {notification}
          </span>
        </div>
        <div>
          <BiX className='h-6 w-6' onClick={removeAlert} />
        </div>
      </div>
    </>
  )
}

export default AlertNotification
