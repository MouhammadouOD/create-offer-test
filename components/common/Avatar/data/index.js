export default {
  src: 'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxMHERURERERERMSExEVEBIQEBIQGhUdFhUWFhUWFRUYHSkgGBomGxMVITEhJSkrLi46Fx8zRDMsNykuLisBCgoKDg0OGxAPGzElHx83LS4uNy0rLS0zLS0tLi0tLS0tNysrKzctLS0tLS0rKy0tLTctLS03KzctNzctNy0rK//AABEIAMgAyAMBIgACEQEDEQH/xAAcAAEAAgMBAQEAAAAAAAAAAAAABgcDBAUCAQj/xAA9EAACAgEDAgQDBgIIBgMAAAABAgADEQQFIRIxBhNBUQdhcRQiMoGRoVKxFSNCgpPB0fAzYnKSlNIkJVP/xAAbAQEAAwADAQAAAAAAAAAAAAAABAUGAQIDB//EACYRAQACAQIFBAMBAAAAAAAAAAABAgMEEQUSEyExFEFCYRZRoQb/2gAMAwEAAhEDEQA/ALxiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiYNRqk0wzY6IPd2C/uTAzxOcN90rdtTR/jJ+veblVy3DKsrD5EH+UDLERAREQEREBERAREQEREBERAREQEREBERA+Tk7/AL9TsSBrjyxC1VJhrLWJAC1pkFjk8+g75E++IN2Gz19QU2WOQlFS/isc5wo9hgEk9gATOPt22fYWOr1ZW3WOOCQCKQc4rpBH3QAeSOTzPLNlripN7z2hzWs2naGArue/sD1LtunIzhcW3nPYHqGEOO4wSM47iblPgfRkf16Nq36i/XqnNzZIAOM4AHHYDHynhtUzd2bn5kfsOJ8quNTBu5HvM/8AkVObaKdkv0dtnZPh7SFej7JpunAGPIr9PymofCOkRemuryOcg6d3pPYjupHHPbtMv9Ncfh549ePnzPQ3ofwn9RLGOL6Wfk8fT5P01Psus2oZqs+2ID/w7yEsxjstoABIxxkc57+s62360a5chXQg4dLFKspHoQffuCMg98zYqfzAD7gH9Zpa3a11DeauEvUdKWgAkDOek/xKT3Esq2i0bx7vJ0omjtupa5SLVCWr+NRyO7BXU/wsFJGeRyPSb07OCIiAiIgIiICIiAiIgIiICIiAnknpnqR7x3q30egvNeA7KK07jmwhBjHrluIGh4f/APvdRZuLg9FZso0QPZVU4stA/idgRnvgAZwTPl+oN7Fj6/7AHyne2/bht+lTTpx5dQQdzyBgnJ5POTI3YprJU8EZ4mY/0Nr7UiPCfoYiZl76o6pizGZldlnysvVHVMWYzBypFser81eknle30nV7yF0ahtOwZTgj/ZBHtO/XvadIJz1eoAJ/Q9v3mw4VxPHOLp5Z2mP6qtTprRbeviWxrKxUwvw2UVlIT+0pIJBGcHBGR6jn3M3VbqGR64/efPxj6/5zDo2ZkHV+IZDY47EgkAdhxxNBE7xvCE2YiJyEREBERAREQEREBERAREQE4/iLazu1aIrhem/T2tkE5FdgcjA9T08Z4nYmKxxUCzEAKCSTgYxySSewgZJztftS6s5z0n3GP3kHHj/UeJdS+m2eiu1aj/W6vUMy1r3GVAGTkg47k98Acz7u28714YQ6i+rS62hObhpw9bovqwBBBAHc8474wCZ5ZcFM1eW8bw7VvNJ3hMK9gRe5Y/XA/kJzt22waIdQbgnGDj9j6/pNzwn4mo8VUC/TsSO1iN+JGxkqw/kRwf5b+66Qausg9xkr+krNXwvBOCYx17wk4tTfnjmnsiGYzMZOI6piZiYnaV7EbsmZ1Nl0B1DByMKD+uO2Plmc3QILrFU9iRn/AE/Pt+cnFaCsYAwBLzg2gjNbqX8VV+tzzjjlj3ZJircMWA/snDfUgN/Iiei3TyeO/wC0j3gSw6jTNc2c36jU2jOfwta3QRn06QuPlibJTpJERAREQEREBERAREQEREBERASuvjruzbXtTKhIOosSnI44IZ2GfmEI/OWLK3+PO2NuG1F0Gfs9yWsBzwAyMfoBZk/QwNj4I7aug2mpgB1XF7HPvk4GfoAB+UnjoLAQQCCCCDznPBBHqJB/gtrl1u0UBcZq663HsVY4z7ZBB/OTlmC8ngDPfjt35gUD8MLj4c3/AFGhUkUvZfWFz/Axas8+oXj35l8ayzykZhzgE/tKF+GtZ8ReIr9YozUlmosDfJmK18+5BE/QDL1DE6ZImazEe7mO0q8Z+o598/vPnVMu50/ZbWX2Jx+fI/YzWXk4nzvJjmMk1ny1VJiaxaHQ2rTNqbF6QcBgSfbBB7+/HAkm1O6hLBUg6mJAPy9T+eJ62XSnSVBWADdzj/M+pmhpNP8AY7LdReQir1HrYgDHckn0AAAmp0mlyafHWtPN/P0pM+WmW9pt8fH2eONe2j0pSv8A42pZaKB/zWHBIHrheon6Tr7Xol26mulfw1IqD+6AP8pFfDXX4n1J3KxSunr6k21GBBIJw+oYHsWxheOAT75M1l8ryIiAiIgIiICIiAiIgIiICIiAmG+hdSrI6hlYEMp5BBGCCPbEzRAq3S+Ctd4Kvst2h67tPac2aPUsUx3x0PwMjsCSD2BzjM2N+p3zxRWdMKNNt1Vg6brftI1DlT+IKVAwCMgjGT2yATLKiBG/BXhGjwdp/JpyzNzda2A1jY7kDsB2AHYepOSZJEjm47br1d30uuQKxyKtTphaFzjIV0ZSBxwCDjJ5gZPEGztryrJgEAg54+hz8uf1nnaPD40p67CGYdh6D5/MzQVd7Ax1bSe/JXVj6HAOB9Jkq2nc9UQb9wqqXI6k0mlAz7gWWMSPqBIM8Pwzm6sx3SI1WSMfTieztbtvFOzp13WBe2FGWZiewVBksT6ACR63bbvFzg6uttPokIZNMWHXeQchrwMhUHcJnPv7Tr7T4X0+1t5iobLT3uudrX/JmJI+gwJ3JOR2OtBWAAMAYAAwAMcAADsJkiICIiAiIgIiICIiAiIgJzN037S7RgajU00k/hWyxFJ+YUnJH0EeJNy/ofSX6jv5NVj/APapI/fEg/wc20a7TNuOoAt1OqsctZYAxVVJVVUkcDA7CBPtt3SndV69PdVcvq1ViuPoSCcH5GfNz3WjalDai+mhWJCtdYlQOBkgFiATj0lZ+PceC9z0Wt0wFS6qwU6qpPurYMqCxUcZAfIOM5HzMmvxFQPteryAR5DnkA9hkHB9cwO3odbXuNYtpsS2ts9L1OtinBIOGUkHBBHf0Mx7nulG0qH1F9VCk9KtdYlQJwTgFiATgE4HPBlVfBXXvsd9+z6g8ri/T9+QyqzdOfQqVbHpzOV8atQ29OXB/wDj6HUVaZQOQ9toL3E+/SERce5PzgXXp9zp1dXn13VWVAMTalisn3c9RLgkADByc8YnOHjPbu39I6L/AMun/wBp0trrFVFaqAqitMAAAD7o4AHAEqHwlqtFte97p9qbTVL1IKvP6FHJJYLnjJyCcfKBa23+INLujdFGq09zgZK0312HHuQpJA+cblv+k2lgmo1WnoYgELdfXUSCSAQGIJGQRn5GcPw3qdDvmstv0YQtplWp7agoWzzQWxwBnp6Bz8yPSR/4/VK+31kqCRqK8EgZGcggHuAfWBO9F4j0etIWrWaawnsK763J+gBJM6gOZFtw8G6PfNKtdmnqUmpeiytFR0PSMFWUAjBwcdpHfg3vl+qGq0OpdrH0NgVbG5JViwAJPJwUOCfQiBMtT4q0Okc12a7SI6khkfU0qwI7gqWyD9RPlPizQXsEXX6NmYgKq6mkliTgBQGyST7SufiSaNLv222XipK+hzc1gULwWALE8HBxgn5STanctq33UU6Wk0XWlxaj6foJr8ohwSwHY9sA+pgS7cdzp2peu+6qhCcBrrFrBOM4BYgZwDxPWg19W5ILKLa7qySA9TrYpwcEBlJBIIx3mj4wrFuh1IYAjyLeCAeykjg/MSrvgxrn8Oau3abzxaqX6UnjOVDMB9VIOPToI7mBbm47nTtSdeouqoQkKHusSpckEgBmIGSATjOeDPe37hVuaeZRbXchJAep1sXjuOpSRke0pf456lt2yFP9Robaq39eu20FiP7qqAf+qXJsdK6fTUqihVFVeFUAD8I7AQNm60UKWYhVUFmYkAAAEkkngAAEk/KfNPcupVXRg6OAyMpBDAgEEEcEEYIkV+I9j6yqvb6W6bdc5r6u/Qijqscgc4wAM/PHrOR8Fd2a3S2bfdkX7fa9RB79PUen9CGX5YAgWRERAREQEREDnb/tw3fTXac8C6p0P94EfzMgPwq3dNhobbNay6fUaZ7MC4isWKxJVkZjhh9DLPmnrtsp3EYupqtx282tX/TqBxAq/wAWMPiBumk02lPmafRWebqr05QHKnoVxwSQoHBPfPpJj8TNZXots1XmOq9dTKgZgOotgAKCck89hJJpdKmjXorRK1HZUUIPyAGI1GnTUDDorgejqG/MAgwKd+IOmfytBvO3MtltK01OUw4PUAFDYPJDMQQe3V6YmL4n6RNg2fSaZ3U3tqq7bR1DqZmFjWOQTnHUSMngcCXPXpkqHSqKF79IUAfXAGM8Tzfo69QcvWjn3dFb9yDAwbXq0u06WK6Mnlr98MpXhRkkg4GMHPPGJVHgXdtIm97o1t2nCWMnlNY6BW6SQ3SxODj5GXClCVr0hVC8/dCgDnvwOJhO10f/AIU/4af6QItoN+0NWu8nQmu67VkNqPJYFUWtWAclRjJyBjPPHbE4Hx+1laaGusuoc6ipgnUOrAzk9PfHB5xjiWXToa9OcpXWp55VFU8+mQMxfoq9QcvWjntl0VvfjJB45P6wItq/Huh2vTKRqK7rPLUV00HzXdukYUKuSDn3xOX8IvD1+3LqdbqlNduusD+WeCqgsVDD0JLnj6SeVaCqk5WqtT7qij9wJtQKd+IW50J4g20vbV0VK4uJZSE6iwAfuBkkd/eSnxB4j2zbmTUrZp7dTV1Lp007oWdrAFCErngkjk8DvJdZt1Nhy1NbE9ya1P5kkcz4u2UqcimoEYwRWvp2IOOIHJ8a7gmi2697mWvqosGCwGWZSAozgk5PYcytvHelOp0O37xoGDXaNaUYphsghcBgCeVYkEH+My5b6F1Aw6q49mUN+eDPlemSpelUUKf7IUAfmAMeggUv8SdKNm2Oqq11+03aqq+8ZHUWYOznHfAyBnGBgCW3sesr1OlqsSxWTyk+8rAgYUZyRwMeue03L9HXqDl60c+7orfzE9V0LUvSqqq8/dCgDnvwOOYFdbYlfjvX6jUpq7q00uNPp/stqoxB5tc5BOC2ACAAQoOZHd0VPh1vtF32h7KtahXUm51ZwSQCzkAcZKEEgdj3xLl0+kr034K0Qnv0Iq5+uAMz5doa7zl6q2PuyKx+mSMwMlVotAZSGBAIZSCDnkEEcEfMTLPCIEGAAAMYAGP0HpPcBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERAREQEREBERA//2Q==',
  alt: 'Demandez nous un conseil',
  size: 10,
  className: ''
}
