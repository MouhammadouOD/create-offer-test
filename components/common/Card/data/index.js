export default {
  children: <div></div>,
  className: 'rounded-2xl'
}
