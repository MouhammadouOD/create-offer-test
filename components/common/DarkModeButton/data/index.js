export default {
  className:
    'w-12 h-6 text-lg flex justify-center items-center bg-omedema-vdark border border-omedema-vdark hover:bg-omedema-dark hover:border-omedema-dark rounded-full cursor-pointer'
}
