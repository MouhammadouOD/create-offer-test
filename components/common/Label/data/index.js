export default {
  children: 'Children',
  colorBg: 'bg-orange-300',
  colorText: 'text-white',
  size: 'small',
  colorOutline: 'outline-blue-200'
}
