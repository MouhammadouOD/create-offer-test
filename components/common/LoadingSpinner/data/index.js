export default {
  message: 'Message',
  displayMessage: true,
  size: 'small'
}
