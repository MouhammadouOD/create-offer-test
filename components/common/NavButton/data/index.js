export default {
  listNav: [
    {
      name: 'name1',
      index: '1'
    }
  ],
  nav: 2,
  onClick: nav => {
    nav = 1
  },
  className: 'text-black'
}
