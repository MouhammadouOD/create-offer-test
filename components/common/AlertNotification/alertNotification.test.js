import React from 'react'
import { testUtils } from '../../../helpers'
import '@testing-library/jest-dom'
import AlertNotification from '.'
import data from './data'

const { render, screen, renderer } = testUtils

describe('AlertNotification', () => {
  it('Should render Snapshot', async () => {
    const component = renderer.create(
      <AlertNotification message={data.message} type={data.type} />
    )
    expect(component).toMatchSnapshot()
  })

  it('Should render with good props', async () => {
    const component = render(
      <AlertNotification message={data.message} type={data.type} />
    )

    const div1 = screen.getByTestId('displayAlert')
    expect(div1).toHaveAttribute(
      'class',
      `fixed bottom-0 right-0 z-50 flex flex-row justify-between items-center cursor-pointer rounded-md px-4 py-3 my-2 md:px-6 md:py-4 md:my-4 mx-auto text-sm md:text-base 2xl:text-lg w-4/5 lg:w-2/5 2xl:w-2/5  bg-red-200 `
    )

    const span1 = screen.getByTestId('typeNotification1')
    expect(span1).toHaveAttribute(
      'class',
      `inline-block mb-1 h-6 w-6 text-red-600`
    )

    const span2 = screen.getByTestId('typeNotification2')
    expect(span2).toHaveAttribute(
      'class',
      `ml-2 inline-block w-11/12 text-red-800`
    )
    expect(span2).toHaveTextContent(data.message)
  })
})
