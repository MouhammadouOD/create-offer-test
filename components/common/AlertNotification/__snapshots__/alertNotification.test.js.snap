// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertNotification Should render Snapshot 1`] = `
<div
  className="fixed bottom-0 right-0 z-50 hidden flex-row justify-between items-center cursor-pointer rounded-md px-4 py-3 my-2 md:px-6 md:py-4 md:my-4 mx-auto text-sm md:text-base 2xl:text-lg w-4/5 lg:w-2/5 2xl:w-2/5  bg-red-200 "
>
  <div
    className="w-full flex flex-row items-center"
  >
    <span
      className="w-1/12"
    >
      <svg
        className="inline-block mb-1 h-6 w-6 text-red-600"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m16.192 6.344-4.243 4.242-4.242-4.242-1.414 1.414L10.535 12l-4.242 4.242 1.414 1.414 4.242-4.242 4.243 4.242 1.414-1.414L13.364 12l4.242-4.242z"
        />
      </svg>
    </span>
    <span
      className="ml-2 inline-block w-11/12 text-red-800
            "
    >
      Message
    </span>
  </div>
  <div>
    <svg
      className="h-6 w-6"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m16.192 6.344-4.243 4.242-4.242-4.242-1.414 1.414L10.535 12l-4.242 4.242 1.414 1.414 4.242-4.242 4.243 4.242 1.414-1.414L13.364 12l4.242-4.242z"
      />
    </svg>
  </div>
</div>
`;
